### PR TITLE
feat(NX-3130) Add tertiary CTA to contact gallery for BNMO artworks

### PR DESCRIPTION
**Pull request by lean8086**
The type of this PR is: **Feature**

This PR solves [NX-3130]

### Description

It might be worth reviewing each individual commit :)

- Converted the `ArtworkSidebarPartnerInfo` component to a FC
- Use the hook `useInquiry` to display the inquiry modal
- Migrated existing tests to RTL

#### Examples

**BN only**

image

**BNMO**

image

**Inquireable (doesn't render the button)**

image

cc @artsy/negotiate-devs 

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NX-3130]: https://artsyproduct.atlassian.net/browse/NX-3130?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPartnerInfo.tsx [hunk 1]
```diff
@@ -37,51 +63,71 @@ export class ArtworkSidebarPartnerInfo extends Component<
...
-      artwork.partner.locations.length > 0 &&
-      // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
-      filterLocations(artwork.partner.locations)
-
-    return (
-      <>
-        <Spacer mt={2} />
-
-        {this.renderPartnerName()}
+  const handleInquiry = () => {
```
Comment (araujobarret) [inline src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPartnerInfo.tsx:67]: checked in the ticket and we're using the same event 👍

**Review by araujobarret — CHANGES_REQUESTED** [review]
Everything looks good, but I noticed the border color of the CTA is different from Figma, can you check that, please?

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPartnerInfo.tsx (hunk 1 above)
Comment (lean8086) [reply src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPartnerInfo.tsx:67]: Good catch! I checked with Peter and corrected the color.

**Comment by lean8086** [comment]
> Everything looks good, but I noticed the border color of the CTA is different from Figma, can you check that, please?

Good catch! I checked with Peter and corrected the color.

**Review by araujobarret — APPROVED** [review]

**Pull request merged**